core/dbt/events/functions.py

```diff
@@ -57,7 +57,11 @@ def setup_event_logger(log_path, level_override=None):
     EVENT_HISTORY = deque(maxlen=flags.EVENT_BUFFER_SIZE)  # type: ignore
 
     make_log_dir_if_missing(log_path)
-    this.format_json = flags.LOG_FORMAT == "json"
+    # Flags is not set in the integration test suite but is by the environment.
+    # DBT_LOG_FORMAT is used in our test suite so checking for it's presence is key.
+    this.format_json = (
+        flags.LOG_FORMAT == "json" or os.environ.get("DBT_LOG_FORMAT", "json") == "json"
```

Comment:
When flags.py doesn't set it (rare but it happens) we must do an ancillary check or else the logging will get messed up by non-json messages referring to a _test database -- some of these are duplicates of subsequent json messages and some appear not to be  (hard to verify)